pkg/endpoint/directory.go

```diff
@@ -81,7 +81,9 @@ func moveNewFilesTo(oldDir, newDir string) error {
 
 	for _, oldFile := range oldFiles {
 		if ok = newFilesHash[oldFile]; !ok {
-			os.Rename(filepath.Join(oldDir, oldFile), filepath.Join(newDir, oldFile))
+			if err := os.Rename(filepath.Join(oldDir, oldFile), filepath.Join(newDir, oldFile)); err != nil {
+				return err
```

Comment:
Could you add some context here like `fmt.Errorf("failed to rename new endpoint state files: %w", err)`

Reply:
Done